--- a/include/ck_tile/host/kernel_launch.hpp	2024-10-01 05:00:50.000000000 +0800
+++ b/include/ck_tile/host/kernel_launch.hpp	2024-11-14 17:12:54.058228000 +0800

@@ -34,7 +34,7 @@
         for(int i = 0; i < s.cold_niters_; ++i)
         {
             kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-            hip_check_error(hipGetLastError());
+            HIP_CHECK_ERROR(hipGetLastError());
         }

         const int nrepeat = s.nrepeat_;
@@ -49,7 +49,7 @@
         for(int i = 0; i < nrepeat; ++i)
         {
             kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-            hip_check_error(hipGetLastError());
+            HIP_CHECK_ERROR(hipGetLastError());
         }

         HIP_CHECK_ERROR(hipEventRecord(stop, s.stream_id_));
@@ -64,12 +64,12 @@
     else
     {
         kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-        hip_check_error(hipGetLastError());
+        HIP_CHECK_ERROR(hipGetLastError());
         return 0;
     }
 #else
     kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-    hip_check_error(hipGetLastError());
+    HIP_CHECK_ERROR(hipGetLastError());
     return 0;
 #endif
 }
@@ -101,7 +101,7 @@
         // warm up
         preprocess();
         kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-        hip_check_error(hipGetLastError());
+        HIP_CHECK_ERROR(hipGetLastError());

         const int nrepeat = 10;
 #if CK_TILE_DEBUG_LOG
@@ -119,7 +119,7 @@
         {
             preprocess();
             kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-            hip_check_error(hipGetLastError());
+            HIP_CHECK_ERROR(hipGetLastError());
         }

         HIP_CHECK_ERROR(hipEventRecord(stop, s.stream_id_));
@@ -135,13 +135,13 @@
     {
         preprocess();
         kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-        hip_check_error(hipGetLastError());
+        HIP_CHECK_ERROR(hipGetLastError());

         return 0;
     }
 #else
     kernel<<<grid_dim, block_dim, lds_byte, s.stream_id_>>>(args...);
-    hip_check_error(hipGetLastError());
+    HIP_CHECK_ERROR(hipGetLastError());

     return 0;
 #endif

--- a/include/ck_tile/host/hip_check_error.hpp    2024-10-01 05:00:50.000000000 +0800
+++ b/include/ck_tile/host/hip_check_error.hpp    2024-11-14 17:36:21.878209000 +0800
@@ -22,6 +22,7 @@
 }
 } // namespace ck_tile

+#ifndef HIP_CHECK_ERROR
 #define HIP_CHECK_ERROR(retval_or_funcall)                                         \
     do                                                                             \
     {                                                                              \
@@ -34,3 +35,4 @@
             throw std::runtime_error(ostr.str());                                  \
         }                                                                          \
     } while(0)
+#endif
